### PR TITLE
Adding Menu Parsing File

### PR DESCRIPTION
**Pull request by tm185221**
## Description
Adds the menu parsing file to the repo. This file is responsible for parsing the menu. The is only a piece of the backend update. The app still needs to be updated to connect the new back and front end together but the flow of the site, needs to happen before that can be implemented.

## Testing
There is nothing to test for this change. The routes of the application were not changed.

## Ticket
MARKET-5092

### CODE REVIEW

File: code/menuParser.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import json
+ 
+file = open('currentMenu.json')
+data = json.load(file)
+file.close()
+
+
```
Comment (co185063) [inline code/menuParser.py:7]: extra line 🐧

File: code/menuParser.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+def getLinkedGroupContents(linkgroup):
+
+
```
Comment (co185063) [inline code/menuParser.py:10]: extra lines 🐧

File: code/menuParser.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+file = open('currentMenu.json')
+data = json.load(file)
+file.close()
+
+
+def getLinkedGroupContents(linkgroup):
+
+   
+    for items in data['linkGroups']:
+
```
Comment (co185063) [inline code/menuParser.py:12]: extra line 🐧

File: code/menuParser.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    for ids in toppingsList['linkedItemReferences']:
+        for items in data['linkedItems']:
+            if ids['posName'] == items['displayName']:
+                   price = items['prices'][0]['price']
+                   priceList.append({'displayName':items['displayName'],'price':price})
+    print(priceList)
+    return priceList
+
+
+
```
Comment (co185063) [inline code/menuParser.py:28]: extra lines 🐧

File: code/menuParser.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+def menuParsing():
+    
+     lunchMenuItems = []
+     groupToppingsList = ['Sandwich-Toppings']
+     #dinnerMenuItems = []
+
+
+
```
Comment (co185063) [inline code/menuParser.py:36]: extra lines 🐧

File: code/menuParser.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        uniqueToppingsWithPrices = []
+        groupToppingsWithPrices = []
+        sharedToppingsWithPrices = []
+           
+
+        if len(item['linkGroupIds'])!= 0:
+
+         
+
```
Comment (co185063) [inline code/menuParser.py:50]: extra lines 🐧

File: code/menuParser.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+     dinnerID = 0
+
+     for item in data['salesItems']:
+
+        uniqueToppingsWithPrices = []
+        groupToppingsWithPrices = []
+        sharedToppingsWithPrices = []
+           
+
+        if len(item['linkGroupIds'])!= 0:
```
Comment (co185063) [inline code/menuParser.py:47]: In this line, I would try to do something like if !len(item['linkGroupIds']): Not sure if this line would work but I prefer to leave out the != 0 in expressions like these

File: code/menuParser.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+
+
+  
+
+
+
+
+    
+
```
Comment (co185063) [inline code/menuParser.py:95]: Extra lines at the end of the file 🐧

File: code/menuParser.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+                    groupToppingsWithPrices = getLinkedGoupContentsIds(groupToppings)
+                
+
+               
+        iteminfo = {"id":lunchID ,"displayName":item['displayName'],'description':item['description'] , 'price': item['currentPrice'],
+                    'tags': item['tags'],'uniqueToppings':uniqueToppingsWithPrices,'sharedToppings':sharedToppingsWithPrices, 'groupToppings':groupToppingsWithPrices}
+        lunchID += 1
+        lunchMenuItems.append((iteminfo))
+
+
```
Comment (co185063) [inline code/menuParser.py:72]: extra lines

File: code/menuParser.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+                    sharedToppingsWithPrices = getLinkedGoupContentsIds(sharedToppings)
+               
+                else:
+                    groupToppings = getLinkedGroupContents(linkgroup)
+                    groupToppingsWithPrices = getLinkedGoupContentsIds(groupToppings)
+                
+
+               
+        iteminfo = {"id":lunchID ,"displayName":item['displayName'],'description':item['description'] , 'price': item['currentPrice'],
+                    'tags': item['tags'],'uniqueToppings':uniqueToppingsWithPrices,'sharedToppings':sharedToppingsWithPrices, 'groupToppings':groupToppingsWithPrices}
```
Comment (co185063) [inline code/menuParser.py:68]: Can we align these two lines?

File: code/menuParser.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def getLinkedGoupContentsIds(toppingsList):
+    priceList = []
+    for ids in toppingsList['linkedItemReferences']:
+        for items in data['linkedItems']:
+            if ids['posName'] == items['displayName']:
+                   price = items['prices'][0]['price']
+                   priceList.append({'displayName':items['displayName'],'price':price})
+    print(priceList)
```
Comment (co185063) [inline code/menuParser.py:24]: Do we need this print statement?

File: code/menuParser.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+               
+        iteminfo = {"id":lunchID ,"displayName":item['displayName'],'description':item['description'] , 'price': item['currentPrice'],
+                    'tags': item['tags'],'uniqueToppings':uniqueToppingsWithPrices,'sharedToppings':sharedToppingsWithPrices, 'groupToppings':groupToppingsWithPrices}
+        lunchID += 1
+        lunchMenuItems.append((iteminfo))
+
+
+     return lunchMenuItems
+
+print(menuParsing())
```
Comment (co185063) [inline code/menuParser.py:75]: Do we need this print statement?

**Review by co185063 — CHANGES_REQUESTED** [review]

File: code/menuParser.py (hunk 7 above)
Comment (tm185221) [reply code/menuParser.py:37]: Here I am checking to see if the item has linkGroups. In the menu.json the linkGroupIds will either contain the linkgroupIds r be blank.

Would a better solution be:
`python
if not item['linkGroupIds]:
`

File: code/menuParser.py (hunk 10 above)
Comment (co185063) [reply code/menuParser.py:54]: This conversation was marked as resolved but I don't see any changes or comments left to why we left it this way. I suggest

```
iteminfo = 
    {
        'id':lunchID ,
        'displayName':item['displayName'],
        'description':item['description'], 
        'price': item['currentPrice'],
        'tags': item['tags'],
        'uniqueToppings':uniqueToppingsWithPrices,
        'sharedToppings':sharedToppingsWithPrices, 
        'groupToppings':groupToppingsWithPrices
    }
```

Or compared to the original line but with a different formatting

```
iteminfo = 
    {
        "id":lunchID ,
        "displayName":item['displayName'],
        'description':item['description'], 
        'price': item['currentPrice'],
        'tags': item['tags'],
        'uniqueToppings':uniqueToppingsWithPrices,
        'sharedToppings':sharedToppingsWithPrices, 
        'groupToppings':groupToppingsWithPrices
    }
```

The differences between the first block of code and the second is that I updated the items id and displayName to use ' ' and not the " ".

File: code/menuParser.py (hunk 7 above)
Comment (co185063) [reply code/menuParser.py:37]: I think that line should work. If you test it and it works def add that in in this line's place

File: code/menuParser.py (hunk 3 above)
Comment (co185063) [reply code/menuParser.py:9]: Extra extra line 🐧

**Review by kg185155 — APPROVED** [review]

**Review by co185063 — APPROVED** [review]
LGTM!

**Pull request merged**